src/Hal/Bundle/PhpMetricsCollector/Collector/PhpMetricsCollector.php

```diff
@@ -1,101 +1,122 @@
+            : $this->kernel->getRootDir() . '/../';
+        $projectDir = realpath($projectDir);
+        $files = array_filter($files, function ($filePath) use ($excludes, $projectDir) {
+            return ! preg_match(
+                '!' . implode('|', $excludes) . '!',
+                mb_substr($filePath, mb_strlen($projectDir))
+            );
         });
-
-        // Prepare datas
-        $all = $average = array(
-            'cfiles' => 0,
-            'maintainability' => array(),
-            'commentWeight' => array(),
-            'complexity' => array(),
-            'loc' => array(),
-            'lloc' => array(),
-            'cloc' => array(),
-            'bugs' => array(),
-            'difficulty' => array(),
-            'intelligentContent' => array(),
-            'vocabulary' => array(),
-        );
-        $scoreByFile = array();
-
-        // group files into tmp folder
-        $folder = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid();
-        mkdir($folder);
-        foreach ($files as $file) {
-            // run PhpMetrics
-            $tokenizer = new Tokenizer();
-            $tokenType = new TokenType();
-
-            // halstead
-            $halstead = new Halstead($tokenizer, $tokenType);
-            $rHalstead = $halstead->calculate($file);
-
-            // loc
-            $loc = new Loc($tokenizer);
-            $rLoc = $loc->calculate($file);
-
-            // complexity
-            $complexity = new McCabe($tokenizer);
-            $rComplexity = $complexity->calculate($file);
-
-            // maintainability
-            $maintainability = new MaintainabilityIndex();
-            $rMaintenability = $maintainability->calculate($rHalstead, $rLoc, $rComplexity);
-
-            // store result
-            $files[$file] = array();
-            $all['cfiles']++;
-            $all['maintainability'][] = $scoreByFile[$file]['maintainability'] = $rMaintenability->getMaintainabilityIndex();
-            $all['commentWeight'][] = $scoreByFile[$file]['commentWeight'] = $rMaintenability->getCommentWeight();
-            $all['complexity'][] = $scoreByFile[$file]['complexity'] = $rComplexity->getCyclomaticComplexityNumber();
-            $all['loc'][] = $scoreByFile[$file]['loc'] = $rLoc->getLoc();
-            $all['lloc'][] = $scoreByFile[$file]['lloc'] = $rLoc->getLogicalLoc();
-            $all['cloc'][] = $scoreByFile[$file]['cloc'] = $rLoc->getCommentLoc();
-            $all['bugs'][] = $scoreByFile[$file]['bugs'] = $rHalstead->getBugs();
-            $all['difficulty'][] = $scoreByFile[$file]['difficulty'] = $rHalstead->getDifficulty();
-            $all['intelligentContent'][] = $scoreByFile[$file]['intelligentContent'] = $rHalstead->getIntelligentContent();
-            $all['vocabulary'][] = $scoreByFile[$file]['vocabulary'] = $rHalstead->getVocabulary();
+        $config = new Config();
+        $config->set('exclude', $excludes);
+        $config->set('files', $files);
```

Comment:
I think that the configuration could be deducted from the constructor directly in order to reduce the responsibility of this method.

* `$excludes` is hardly defined in this method, so it could be easily moved
* `$files` id deducted from `get_included_files()`, `$excludes` and `$projectDir`
  * `$projectDir` is deducted from `$this->kernel`, which is already in the construct.